src/Microsoft.OData.Core/ODataUtils.cs

```diff
@@ -301,5 +301,31 @@ public static string AppendDefaultHeaderValue(string headerName, string headerVa
 
             return extendedMediaType.ToText(encoding);
         }
+
+        /// <summary>
+        /// Construct Url from string form with HTTP version validation.
+        /// </summary>
+        /// <param name="value">The Uri string.</param>
+        /// <returns>The Uri object.</returns>
+        public static Uri BuildRequestUri(string value)
+        {
+            string urlSegment = value;
+            int lastSpaceIndex = value.IndexOf(' ');
+            if (lastSpaceIndex != -1)
+            {
+                // Validate HTTP version
+                urlSegment = value.Substring(0, lastSpaceIndex);
+                string httpVersionSegment = value.Substring(lastSpaceIndex + 1);
+
+                if (string.CompareOrdinal(ODataConstants.HttpVersionInBatching, httpVersionSegment) != 0)
+                {
+                    throw new ODataException(Strings.ODataBatchReaderStream_InvalidHttpVersionSpecified(
+                        httpVersionSegment,
+                        ODataConstants.HttpVersionInBatching));
+                }
+            }
+
+            return new Uri(urlSegment, UriKind.RelativeOrAbsolute);
+        }

```

Comment:
why are we doing this validation? what if the service uses https?  or http 1.0?
We should just be validating the odata batching semantics and not get in the way of whatever other protocol they may be trying to use.

Reply:
OK, removed the validation part.

BTW, It was done so to follow the same pattern on multipart batch side:
[https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/ODataBatchReader.cs#L573](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/ODataBatchReader.cs#L573)
but I think we don't have to do this on JSON batch side.

---
In reply to: [156807126](https://github.com/OData/odata.net/pull/992#discussion_r156807126) [](ancestors = 156807126)